src/G2/Execution/InstTypes.hs

```diff
@@ -42,13 +43,15 @@ newType ng i te =
     (ty, te', ng''')
     
 -- | We want to find all the type variable from curr_expr and replace them with symbolic variable
-instType :: ASTContainer t Type => NameGen -> State t -> (NameGen, State t)
-instType ng st = 
+instType :: ASTContainer t Type => Bindings -> State t -> (Bindings, State t)
+instType b st = 
     let 
-        is = tyVarIds $ curr_expr st
+        ng = name_gen b 
```

Comment:
I'm fine with this as it is currently written, so no need to change it, but a useful bit of syntax to make sure you know is:
```
instType b@(Bindings { name_gen = ng, input_names = is })
```
to have `b` as the entirety of the Bindings, `ng` as the NameGen, and `is` as the input names.  (The use of the `@` is called an "as pattern".)